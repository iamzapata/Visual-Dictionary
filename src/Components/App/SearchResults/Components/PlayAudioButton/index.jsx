import React from "react"
import { string } from "prop-types"
import playAudio from "utils/audio/playAudio"

const PlayAudioButton = ({ audioFile }) => {
  return <button onClick={() => playAudio(audioFile)}>&#9658;</button>
}

PlayAudioButton.propTypes = {
  audioFile: string.isRequired
}

export default PlayAudioButton
