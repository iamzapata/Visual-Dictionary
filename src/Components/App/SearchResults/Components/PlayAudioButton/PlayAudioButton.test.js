import React from "react"
import PlayAudioButton from "./index"
import { render, cleanup, fireEvent } from "react-testing-library"
import playAudio from "utils/audio/playAudio"
jest.mock("utils/audio/playAudio", () => jest.fn())

describe("<PlayAudioButton />", () => {
  afterEach(cleanup)
  beforeEach(() => {
    playAudio.mockClear()
  })

  const createProps = () => ({
    audioFile: "fileToPlay"
  })

  it("Should render self", () => {
    const { container } = render(<PlayAudioButton {...createProps()} />)

    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("Should display a play icon", () => {
    const { queryByText } = render(<PlayAudioButton {...createProps()} />)

    const playIcon = queryByText("►")

    expect(playIcon).toBeTruthy()
  })

  it("Should play audio when button clicked", () => {
    const props = createProps()
    const { queryByText } = render(<PlayAudioButton {...props} />)

    const playButton = queryByText("►")
    fireEvent.click(playButton)

    expect(playAudio).toHaveBeenCalledTimes(1)
  })
})
