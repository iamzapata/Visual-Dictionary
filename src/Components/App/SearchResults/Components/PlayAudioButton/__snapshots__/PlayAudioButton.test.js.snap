// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayAudioButton /> Should render self 1`] = `
<button>
  â–º
</button>
`;
